Source/WebCore/platform/graphics/ca/PlatformCALayerDelegatedContents.h

```diff
@@ -29,33 +29,24 @@
 #include <wtf/MachSendRight.h>
 #include <wtf/ObjectIdentifier.h>
 
-OBJC_PROTOCOL(MTLSharedEvent);
-
 namespace WebCore {
 class IOSurface;
 
-namespace Detail {
-enum PlatformCALayerDelegatedContentsIdentifier { };
-}
-
-struct PlatformCALayerDelegatedContentsFinishedEvent {
-    MachSendRight sharedEvent;
-};
-
-struct PlatformCALayerInProcessDelegatedContentsFinishedEvent {
-    id<MTLSharedEvent> sharedEvent;
+class WEBCORE_EXPORT PlatformCALayerDelegatedContentsFence : public ThreadSafeRefCounted<PlatformCALayerDelegatedContentsFence> {
+public:
+    PlatformCALayerDelegatedContentsFence();
+    virtual ~PlatformCALayerDelegatedContentsFence();
+    virtual bool waitFor(Seconds) = 0;
 };
 
 struct PlatformCALayerDelegatedContents {
-    using FinishedIdentifier = ObjectIdentifier<Detail::PlatformCALayerDelegatedContentsIdentifier>;
     MachSendRight surface;
-    FinishedIdentifier finishedIdentifier;
+    RefPtr<PlatformCALayerDelegatedContentsFence> finishedFence;
 };
 
 struct PlatformCALayerInProcessDelegatedContents {
```

Comment:
Is there a build configuration where both `PlatformCALayerDelegatedContents` and `PlatformCALayerInProcessDelegatedContents` are required in the same binary?

Reply:
Yeah, WebGL GPUP is toggleable on runtime, as is ui-side compositing